packages/expect/src/jestMatchersObject.ts

```diff
@@ -59,19 +54,20 @@ export const setMatchers = (
       // expect is defined
 
       class CustomMatcher extends AsymmetricMatcher {
-        sample: Array<any>;
+        sample: Array<unknown>;
 
-        constructor(inverse: boolean = false, ...sample: Array<any>) {
+        constructor(inverse: boolean = false, ...sample: Array<unknown>) {
           super();
           this.inverse = inverse;
           this.sample = sample;
         }
 
-        asymmetricMatch(other: any) {
-          const {pass} = ((matcher(
-            (other: any),
-            ...(this.sample: any),
-          ): any): SyncExpectationResult);
+        asymmetricMatch(other: unknown) {
+          // @ts-ignore
+          const {pass}: ExpectationResult = matcher(
```

Comment:
Would type-casting the result of the matcher call to `SyncExpectationResult` work in this case?